pkg/mons/mon-endpoints.go

```diff
@@ -0,0 +1,46 @@
+/*
+Copyright 2023 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package mons
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+
+	"github.com/kubectl-rook-ceph/pkg/k8sutil"
+	log "github.com/sirupsen/logrus"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const monConfigMap = "rook-ceph-mon-endpoints"
+
+func GetMonEndpoint(clusterNamespace string) {
```

Comment:
Let's add a unit test for this method so we can establish the unit tests from the start. You'll just need to have the kube client as a parameter to the method.

Reply:
this method is not returning anything, how we should write the unit test?

Reply:
As discussed, how about refactoring lines 40-44 to a separate method that can be unit tested.

Reply:
added a unit-test